python_modules/libraries/dagster-slack/dagster_slack/resources.py

```diff
@@ -1,24 +1,59 @@
-from dagster import Field, StringSource, resource
+from dagster import resource
+from dagster._config.structured_config import ConfigurableResource
+from pydantic import Field as PyField
 from slack_sdk.web.client import WebClient
 
 
+class SlackResource(ConfigurableResource):
+    """This resource is for connecting to Slack.
+
+    The resource object is a `slack_sdk.WebClient`.
```

Comment:
I don't think we should use the term "resource object" as that isn't a term of art in the system now.